shared/chat/conversation/messages/popup.desktop.js

```diff
@@ -0,0 +1,53 @@
+// @flow
+import React from 'react'
+import {Box, Button, Icon, PopupMenu, Text} from '../../../common-adapters'
+import {globalStyles, globalMargins, globalColors} from '../../../styles'
+import * as Constants from '../../../constants/chat'
+import {timestampToString} from '../../../constants/chat'
+import type {Message, TextMessage} from '../../../constants/chat'
+
+import type {Props} from './popup'
+import type {MenuItem} from '../../../common-adapters/popup-menu'
+
+function iconNameForDeviceType (deviceType: string): string {
+  switch (deviceType) {
+    case 'mobile': return 'icon-fancy-encrypted-phone-122-x-64'
+    default: return 'icon-fancy-encrypted-computer-150-x-64'
+  }
+}
+
+const TextMessagePopup = ({message}: {message: TextMessage}) => {
+  const {deviceName, deviceType, timestamp} = message
```

Comment:
if you're going to pull it out why not put it into the params above instead of {message} ?